src/spaces/openstreetmap_new.jl

```diff
@@ -0,0 +1,944 @@
+export OpenStreetMapSpace, OSMSpace, OSM
+
+"""
+    OSM
+Submodule for functionality related to `OpenStreetMapSpace`.
+See the docstring of the space for more info.
+"""
+module OSM # OpenStreetMap
+using Agents
+using LightOSM
+using Graphs
+using Pkg.Artifacts
+using LinearAlgebra: dot, norm
+using DataStructures
+using Downloads
+
+export test_map,
+    random_road_position,
+    plan_route!,
+    distance,
+    road_length,
+    random_route!,
+    latlon,
+    intersection,
+    road
+
+
+# Stores information about an agent's path
+struct OpenStreetMapPath
+    route::Vector{Int}      # node IDs along path from `start` to `dest`
+    start::Tuple{Int,Int,Float64} # Initial position of the agent
+    dest::Tuple{Int,Int,Float64}    # Destination. `dest[1] == dest[2]` if this is an intersection
+    return_route::Vector{Int}   # node IDs along path from `dest` to `start`
+    has_to_return::Bool
+end
+
+# NOTE: All positions are indexed by vertex number and _not_ node id
+
+"""
+    OpenStreetMapSpace(path::AbstractString; kwargs...)
+Create a space residing on the Open Street Map (OSM) file provided via `path`.
+The functionality related to Open Street Map spaces is in the submodule `OSM`.
+
+This space represents the underlying map as a *continuous* entity choosing accuracy over
+performance. The map is represented as a graph, consisting of nodes connected by edges. Nodes
+are not necessarily intersections, and there may be multiple nodes on a road joining two
+intersections. The length of an edge between two nodes is specified in the units of the
+map's `weight_type` as listed in the documentation for
+[`LightOSM.OSMGraph`](https://deloittedigitalapac.github.io/LightOSM.jl/docs/types/#LightOSM.OSMGraph).
```

Comment:
I think it is worth metioning here all possible types of `weight_type`, even if it is copy pasting from the OSMGraph docs.